src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -0,0 +1,265 @@
+        }
+
+        /// <summary>
+        /// Function that creates a file to store the broadcast value in the given path.
+        /// </summary>
+        /// <param name="tempDir">Path where file is to be created</param>
+        /// <param name="value">Broadcast value to be written to the file</param>
+        private void WriteBroadcastValueToFile(string tempDir, object value)
+        {
+            Directory.CreateDirectory(tempDir);
+            using FileStream f = File.Create(_path);
+            Dump(value, f);
+        }
+
+        /// <summary>
+        /// Function that serializes and stores the object passed to the given Stream.
+        /// </summary>
+        /// <param name="value">Serializable object</param>
+        /// <param name="stream">Stream to which the object is serialized</param>
+        private void Dump(object value, Stream stream)
+        {
+            var formatter = new BinaryFormatter();
+            formatter.Serialize(stream, value);
+        }
+
+        /// <summary>
+        /// Calls the necessary functions to create org.apache.spark.broadcast.Broadcast object
+        /// for Spark versions 2.3.0 and 2.3.1 and returns the JVMObjectReference object.
+        /// </summary>
+        /// <param name="tempDir">Path where file is to be created</param>
+        /// <param name="sparkContext">JVMObjectReference object of SparkContext</param>
+        /// <param name="value">Broadcast value of type object</param>
+        /// <returns></returns>
+        private JvmObjectReference CreateBroadcast_V2_3_1_Below(
+            string tempDir,
+            JvmObjectReference sparkContext,
+            object value)
+        {
+            WriteBroadcastValueToFile(tempDir, value);
+            var javaSparkContext = (JvmObjectReference)sparkContext.Jvm.CallStaticJavaMethod(
+                "org.apache.spark.api.java.JavaSparkContext",
+                "fromSparkContext",
+                sparkContext);
+            return (JvmObjectReference)sparkContext.Jvm.CallStaticJavaMethod(
+                "org.apache.spark.api.python.PythonRDD",
+                "readBroadcastFromFile",
+                javaSparkContext,
+                _path);
+        }
+
+        /// <summary>
+        /// Calls the necessary Spark functions to create org.apache.spark.broadcast.Broadcast
+        /// object for Spark versions 2.3.2 and above, and returns the JVMObjectReference object.
+        /// </summary>
+        /// <param name="sc">SparkContext object</param>
+        /// <param name="tempDir">Path where file is to be created</param>
+        /// <param name="sparkContext">JVMObjectReference object of SparkContext</param>
+        /// <param name="value">Broadcast value of type object</param>
+        /// <returns></returns>
+        private JvmObjectReference CreateBroadcast_V2_3_2_Above(
+            SparkContext sc,
+            string tempDir,
+            JvmObjectReference sparkContext,
```

Comment:
why do you need to pass this on top of `SparkContext sc`?

Reply:
Removed the unnecessary parameter.